test/integration/crcsuite/prepare.go

```diff
@@ -67,9 +68,14 @@ func CopyFilesToTestDir() {
 	}
 
 	l := strings.Split(cwd, string(filepath.Separator))
-	dataDirPieces := append([]string{string(filepath.Separator)}, l[:len(l)-3]...)
-	dataDirPieces = append(dataDirPieces, "testdata")
+	dataDirPieces := append(l[:len(l)-3], "testdata")
+	var volume string
+	if runtime.GOOS == "windows" {
+		volume = filepath.VolumeName(cwd)
+		dataDirPieces = dataDirPieces[1:] // drop volume from list of dirs
+	}
 	dataDir := filepath.Join(dataDirPieces...)
+	dataDir = fmt.Sprintf("%s%c%s", volume, filepath.Separator, dataDir) // prepend volume back
```

Comment:
Yet another alternative on top of the current code is to set `volume` to `filepath.Separator` in the linux case, and to use `dataDir := filepath.Join(volume, dataDirPieces...)`

Reply:
@teuf , since `filepath.Join` refuses to take two arguments: `string` and `[]string`, it seems that we have a choice between these two code blocks
```
if runtime.GOOS != "windows" {
	volume = string(filepath.Separator)
}
dataDirPieces = append([]string{volume}, dataDirPieces...)
dataDir := filepath.Join(dataDirPieces...)
```
and the current
```
if runtime.GOOS == "windows" {
	volume = filepath.VolumeName(cwd)
	dataDirPieces = dataDirPieces[1:]
}
It would be cleaner if we didn't need L4 in the first block, but I couldn't do it without first prepending `volume` to `dataDirPieces`. Since it's not cleaner the other way, I'd just leave it as it is and merge it (it's tested on all platforms). Unless I'm missing some shortcut! WDYT? 
dataDir := filepath.Join(dataDirPieces...)
dataDir = fmt.Sprintf("%s%c%s", volume, filepath.Separator, dataDir)
```

Reply:
> since `filepath.Join` refuses to take two arguments: `string` and `[]string`,

Did you try with `filepath.Separator(volume, dataDirPieces...)` (3 trailing dots)? This should 'convert' the `[]string` to something appropriate  for `filepath.Join()`

Reply:
You meant `filepath.Join(volume, dataDirPieces...)`? Here's what I had:
```
l := strings.Split(cwd, string(filepath.Separator))
dataDirPieces := append(l[:len(l)-3], "testdata")

var volume string
if runtime.GOOS != "windows" {
	volume = string(filepath.Separator)
}
dataDir := filepath.Join(volume, dataDirPieces...)
```
and here's what I got when I ran `make integration` with it
```
test/integration/crcsuite/prepare.go:78:26: too many arguments in call to filepath.Join
	have (string, []string...)
	want (...string)
```

Reply:
It looks like `filepath.Join` is not happy with a "loose bunch of arguments + 1", but is happy with a "loose bunch of arguments".

Reply:
Indeed, I did not expect that ;) 
Reproducible with
```
package main

import (
	"fmt"
	"path/filepath"
)

func main() {
       a := []string{"a", "b"}
       b := "c"
	fmt.Println(filepath.Join(b, a...))
}
```